### PR TITLE
Update functions to 3.0

### PR DESCRIPTION
**Pull request by ancaantochi**
Update functions binding and sample to work with functions runtime 3.0

Note: 
- nuget package need to be release for bindings with the next release
- vscode extension should be update to create the project with new version

### CODE REVIEW

File: builds/misc/images-release.yaml [hunk 1]
```diff
@@ -19,7 +19,6 @@ jobs:
             docker login '$(build.registry.address)' --username '$(build.registry.user)' --password '$(build.registry.password)'
             docker login '$(registry.address)' --username '$(registry.user)' --password '$(registry.password)'
         displayName: 'Docker Login'
-      - template: ../templates/install-dotnet2.yaml
```
Comment (damonbarry) [inline builds/misc/images-release.yaml:22]: Yay!

Comment (damonbarry) [reply builds/misc/images-release.yaml:22]: There are other YAML files that use this template too:
- builds/checkin/dotnet.yaml
- builds/ci/dotnet.yaml
- builds/misc/images.yaml

Should we remove it from these as well (and remove the template)?

**Review by damonbarry — CHANGES_REQUESTED** [review]
@ancaantochi You still need to remove the reference to `install-dotnet2.yaml` from `builds/checkin/dotnet.yaml`...it's causing the "dotnet Checkin" pipeline to fail.

**Review by damonbarry — APPROVED** [review]

**Review by damonbarry — APPROVED** [review]

File: builds/misc/images-release.yaml (hunk 1 above)
Comment (ancaantochi) [reply builds/misc/images-release.yaml:22]: tests are failing to run after I removed dotnet2 install, I am still trying to figure out why

**Review by damonbarry — APPROVED** [review]

**Pull request merged**